### PR TITLE
Make host orchestrator client usable outside of the cloud orchestrator context

### PR DESCRIPTION
**Pull request by jemoreira**

### CODE REVIEW

File: pkg/client/host_orchestrator_client.go [hunk 1]
```diff
@@ -39,6 +39,7 @@ type HostOrchestratorService interface {
 
 	// Uploads user files to a previously created directory.
 	UploadFiles(uploadDir string, filenames []string) error
+	UploadFilesWithOptions(uploadDir string, filenames []string, options UploadOptions) error
```
Comment (ser-io) [inline pkg/client/host_orchestrator_client.go:42]: nit: empty line above.

File: pkg/client/httputils.go [hunk 2]
```diff
@@ -210,13 +192,20 @@ type fileInfo struct {
 	TotalChunks int
 }
 
+type UploadOptions struct {
+	InitialDuration     time.Duration
```
Comment (ser-io) [inline pkg/client/httputils.go:196]: let's group these 4 fields using a new struct, i.e `BackOffOpts` as they are meant to be used together.

File: pkg/client/host_orchestrator_client.go (hunk 1 above)
Comment (jemoreira) [reply pkg/client/host_orchestrator_client.go:43]: Done

File: pkg/client/httputils.go (hunk 2 above)
Comment (jemoreira) [reply pkg/client/httputils.go:196]: Done

**Review by ser-io — APPROVED** [review]

**Pull request merged**